### PR TITLE
[tools] add PADDLE_API check file diff approvals

### PR DESCRIPTION
**Pull request by gouzil**
### PR types

Others
### PR changes

Others
### Description

修改包含`PADDLE_API`内容的文件都需要`approvals`

相关链接:
 * https://github.com/PaddlePaddle/community/pull/529
 * https://github.com/PaddlePaddle/Paddle/issues/53172

### CODE REVIEW

File: tools/check_file_diff_approvals.sh [hunk 1]
```diff
@@ -367,6 +367,19 @@ if [ "${PHI_INCLUDE_THIRD_PARTY_FILES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; the
     check_approval 1 jiahy0825 zyfncg chenwhql YuanRisheng heavyrain-lzy
 fi
 
+HAS_MODIFIED_PADDLE_API_FILES=`git diff --name-only upstream/$BRANCH | grep "paddle/.*\.h" || true`
+INCLUDE_PADDLE_API_FILES=""
+for CHANGE_FILE in ${HAS_MODIFIED_PHI_HEADER_FILES}; do
+    PADDLE_API_ADDED_LINES=`git diff -U0 upstream/$BRANCH -- ${PADDLE_ROOT}/${CHANGE_FILE} | grep -E "PADDLE_API" || true`
```
Comment (Liyulingyue) [inline tools/check_file_diff_approvals.sh]:
```suggestion
    PADDLE_API_ADDED_LINES=`git diff -U0 upstream/$BRANCH -- ${PADDLE_ROOT}/${CHANGE_FILE} | grep -w "PADDLE_API" || true`
```

File: tools/check_file_diff_approvals.sh [hunk 2]
```diff
@@ -367,6 +367,19 @@ if [ "${PHI_INCLUDE_THIRD_PARTY_FILES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; the
...
+HAS_MODIFIED_PADDLE_API_FILES=`git diff --name-only upstream/$BRANCH | grep "paddle/.*\.h" || true`
+INCLUDE_PADDLE_API_FILES=""
+for CHANGE_FILE in ${HAS_MODIFIED_PHI_HEADER_FILES}; do
+    PADDLE_API_ADDED_LINES=`git diff -U0 upstream/$BRANCH -- ${PADDLE_ROOT}/${CHANGE_FILE} | grep -w "PADDLE_API" || true`
+    if [ "${PADDLE_API_ADDED_LINES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; then
+        INCLUDE_PADDLE_API_FILES="${INCLUDE_PADDLE_API_FILES} ${CHANGE_FILE}"
+    fi
+done
+if [ "${INCLUDE_PADDLE_API_FILES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; then
+    echo_line="You must have one RD (sunzhongkai588/SunZhongKai, Ligoml/LiMengLiu, jiahy0825) approval for PADDLE_API. Need to confirm if additional descriptions are needed\n"
```
Comment (jiahy0825) [inline tools/check_file_diff_approvals.sh]: Maybe this message would be better~
```
echo_line="You must have one RD (jiahy0825, zyfncg, chenwhql) or PM (sunzhongkai588, Ligoml) approval for code changes about PADDLE_API. If you add a new PADDLE_API, please make sure you have written detailed comments about the parameter and usage of this PADDLE_API .\n"
```
P.S. Do not forget to change the `check_approval` lines below.

Comment (Liyulingyue) [inline tools/check_file_diff_approvals.sh]:
```suggestion
    echo_line="You must have one RD (jiahy0825, zyfncg, chenwhql) or PM (sunzhongkai588, Ligoml) approval for code changes about PADDLE_API. If you add a new PADDLE_API, please make sure you have written detailed comments about the parameter and usage of this PADDLE_API .\n"
```

File: tools/check_file_diff_approvals.sh [hunk 3]
```diff
@@ -367,6 +367,19 @@ if [ "${PHI_INCLUDE_THIRD_PARTY_FILES}" != "" ] && [ "${GIT_PR_ID}" != "" ]; the
...
+    check_approval 1 sunzhongkai588/SunZhongKai Ligoml/LiMengLiu jiahy0825
```
Comment (Liyulingyue) [inline tools/check_file_diff_approvals.sh]:
```suggestion
    check_approval 1 jiahy0825 zyfncg chenwhql sunzhongkai588 Ligoml
```

**Review by jiahy0825 — APPROVED** [review]
LGTM

**Review by jiahy0825 — APPROVED** [review]
LGTM

**Pull request merged**